src/Appwrite/Utopia/Request.php

```diff
@@ -180,4 +182,28 @@ public function getHeader(string $key, string $default = ''): string
         $headers = $this->getHeaders();
         return $headers[$key] ?? $default;
     }
+
+    /**
+    * Get User Agent
+    *
+    * Method for getting User Agent. Preferring forwarded agent for privileged users; otherwise returns default.
+    *
+    * @param  string  $default
+    * @return string
+    */
+    public function getUserAgent(string $default = ''): string
+    {
+        $forwardedUserAgent = $this->getHeader('x-forwarded-user-agent');
+        if (!empty($forwardedUserAgent)) {
+            $roles = Authorization::getRoles();
+            $isPrivilegedUser = Auth::isPrivilegedUser($roles);
```

Comment:
A privileged user is a Console user. We shouldn't allow console users to override the user agent because then they can spoof requests to the Console.